src/eval_iter_loop.ipynb

```diff
@@ -0,0 +1,323 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# installation\n",
+    "!docker compose version >/dev/null\n",
+    "!docker info >/dev/null\n",
+    "!pip install -r ../requirements.txt"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# If you are using a MacBook with a M1 processor you have to make sure to set the docker default platform to linux/amd64\n",
+    "import os\n",
+    "os.environ[\"DOCKER_DEFAULT_PLATFORM\"]=\"linux/amd64\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Make sure to have your weaviate vectorDB running\n",
+    "!docker compose -f weaviate/docker-compose.yaml up --detach"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import weaviate\n",
+    "local_weaviate_client = weaviate.Client(\"http://localhost:8080\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# imports\n",
+    "from fondant.pipeline import ComponentOp, Pipeline\n",
+    "from pathlib import Path\n",
+    "\n",
+    "from fondant.pipeline.runner import DockerRunner\n",
+    "import pipeline_index, pipeline_eval"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Single run evaluation"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# get Host IP address\n",
+    "import socket\n",
+    "\n",
+    "def get_host_ip():\n",
```

Comment:
Did you try localhost here? Is it also not working for you with the regular pipeline?

Reply:
With the regular pipeline, I have to write host.docker.internal (I put Matthias explanation below). Within the notebook, neither the localhost or the host.docker.internal work. It worked for me with the Host IP address. I've discussed that with @GeorgesLorre and Jan during yesterday or the day before standup and agreed we could provide a function to extract the host ip. 
<img width="632" alt="image" src="https://github.com/ml6team/fondant-usecase-RAG/assets/117269156/faca98fc-0dff-478f-92b0-5c051eb19fa0">

Reply:
I don't understand what the difference is in running it from the notebook. It will still run inside a docker container on your local machine, so I would still expect `http://host.docker.internal` to work.

What OS are you on?

Reply:
That's what I thought as well. I'm on MacOS.